system/jlib/jtrace.cpp

```diff
@@ -768,9 +768,6 @@ class CSpan : public CInterfaceOf<ISpan>
         if (span == nullptr)
             return false;
 
-        if (!span->IsRecording()) //if not sampled, we shouldn't consider this valid?
```

Comment:
maybe outside the scope of this issue, but found this causes a hang under certain parentbased configurations -- further analysis is needed, but removing this for now

Reply:
@jakesmith this amended commit includes the code review changes you requested and approved + this change